# CAR-SALES dataset

Dữ liệu `car-sales.csv`: bộ dữ liệu chứa thông tin các ôtô cũ ở một cửa hàng bán ô tô cũ. 

Bạn hãy thực hiện các yêu cầu sau:


### 1. Đọc dữ liệu và cho biết dữ liệu có bao nhiêu dòng, bao nhiêu cột

Đáp án: (10 dòng, 5 cột)

In [2]:
# Your code here
import pandas as pd
import numpy as np
car = pd.read_csv('data/car-sales.csv', header=0)
car.head(10)

Manufacturer Colour  Odometer  Doors    Price
0       Toyota  White    150043      4   4000.0
1        Honda    Red     87899      4   5000.0
2       Toyota   Blue     32549      3   7000.0
3          BMW  Black     11179      5  22000.0
4       Nissan  White    213095      4   3500.0
5       Toyota  Green     99213      4   4500.0
6        Honda   Blue     45698      4   7500.0
7        Honda   Blue     54738      4   7000.0
8       Toyota  White     60000      4   6250.0
9       Nissan  White     31600      4   9700.0

In [5]:
car = pd.read_csv("data/car-sales.csv", header = 0)
car.head(10)

Manufacturer Colour  Odometer  Doors    Price
0       Toyota  White    150043      4   4000.0
1        Honda    Red     87899      4   5000.0
2       Toyota   Blue     32549      3   7000.0
3          BMW  Black     11179      5  22000.0
4       Nissan  White    213095      4   3500.0
5       Toyota  Green     99213      4   4500.0
6        Honda   Blue     45698      4   7500.0
7        Honda   Blue     54738      4   7000.0
8       Toyota  White     60000      4   6250.0
9       Nissan  White     31600      4   9700.0

### 2. Cho biết thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng

In [7]:
# Your code here
car.dtypes
# thuộc tính phân loại: Manufacturer, Colour
# thuộc tính định lượng: Odometer, Doors, Price

Manufacturer     object
Colour           object
Odometer          int64
Doors             int64
Price           float64
dtype: object

### 3. Trong dữ liệu có thông tin của bao nhiêu xe?

Đáp án: 10 xe

In [13]:
# Your code here
car.shape[0]
# Hoặc
print(len(car))

10


### 4. Đếm số xe theo theo hãng

Đáp án: Toyota: 10 xe, Honda: 3 xe, Nissan: 2 xe, BMW: 1 xe

In [30]:
# Your code here
car['Manufacturer'].value_counts()
# cách 2
# car_sum_manu = car.groupby('Manufacturer').size()
# print(car_sum_manu)

Manufacturer
Toyota    4
Honda     3
Nissan    2
BMW       1
Name: count, dtype: int64

### 5. Đếm số xe theo màu


In [14]:
# Your code here
car['Colour'].value_counts()

White    4
Blue     3
Black    1
Green    1
Red      1
Name: Colour, dtype: int64

### 6. Tổng giá trị các xe trong cửa hàng

Đáp án: 76450 USD


In [22]:
# Your code here
car['Price'].sum()

76450.0

### 7. Top 3 xe có giá cao nhất


In [31]:
# Your code here
car.sort_values(by = 'Price', ascending=False).head(3)

Manufacturer Colour  Odometer  Doors    Price
3          BMW  Black     11179      5  22000.0
9       Nissan  White     31600      4   9700.0
6        Honda   Blue     45698      4   7500.0

### 8. Top 3 xe đã chạy nhiều cây số (km)nhất (Odometer)

Đáp án:
- Xe Nissan, White, 213095 km
- Xe Toyota, White, 150043 km
- Xe Toyota, Green, 99213 km

In [75]:
# Your code here
car = car.sort_values(by = 'Odometer', ascending=False)
car[['Manufacturer', 'Odometer']].head()

Manufacturer  Odometer
4       Nissan    213095
0       Toyota    150043
5       Toyota     99213
1        Honda     87899
8       Toyota     60000

In [57]:
a = car[car.Manufacturer == 'Honda']
d = a.shape[0]
b = car.shape[0]
print(d/b)
c = a['Price'].sum()
print(c/d)
e = a['Colour'].unique()
print(e)

0.3
6500.0
['Red' 'Blue']


### 9. Khảo sát xe của hãng Honda

- Tính tỷ lệ xe Honda trong dữ liệu của cửa hàng (0.3)
- Cho biết giá trung bình của xe hãng Honda của cửa hàng (6500)
- Cho biết các xe của hãng Honda trong cửa hàng có những màu sắc nào (Red, Blue)

In [58]:
# Your code here
a = car[car.Manufacturer == 'Honda']
print(len(a)/len(car))
print(a['Price'].mean())
c = car
e = a['Colour'].unique()
print(e)

0.3
6500.0
['Red' 'Blue']


# 10. Tính toán các giá trị sau của  thuộc tính giá ('Price')

- Giá trung bình các xe (7645)
- Giá trị trung vị các xe (6625)
- Miền giá trị của xe (18500)
- Miền phân vị của xe (2750)
- Điểm phân vị 0.4 và 0.6 của giá ([5750, 7000])

In [59]:
# Tính giá trung bình
mean_price = car['Price'].mean()

# Tính giá trị trung vị
median_price = car['Price'].median()

# Tính miền giá trị
range_price = car['Price'].max() - car['Price'].min()

# Tính miền phân vị
quantile_range = car['Price'].quantile(0.75) - car['Price'].quantile(0.25)

# Tính điểm phân vị 0.4 và 0.6
quantile_40 = car['Price'].quantile(0.4)
quantile_60 = car['Price'].quantile(0.6)

print(f"Giá trung bình: {mean_price}")
print(f"Giá trị trung vị: {median_price}")
print(f"Miền giá trị: {range_price}")
print(f"Miền phân vị: {quantile_range}")
print(f"Điểm phân vị 0.4: {quantile_40}")
print(f"Điểm phân vị 0.6: {quantile_60}")

Giá trung bình: 7645.0
Giá trị trung vị: 6625.0
Miền giá trị: 18500.0
Miền phân vị: 2750.0
Điểm phân vị 0.4: 5750.0
Điểm phân vị 0.6: 7000.0


In [29]:
car['Price'].describe()

count       3.0
mean     4000.0
std       500.0
min      3500.0
25%      3750.0
50%      4000.0
75%      4250.0
max      4500.0
Name: Price, dtype: float64

---

# TIPS dataset

Dữ liệu chứa trong file `tips.csv` chứa thông tin tiền tip của khách hàng tại một nhà hàng. Dữ liệu có các thông tin sau:

- `total_bill`: tổng hoá đơn (USD)
- `tip`: tiền tip/ tiền boa (USD)
- `sex`: giới tính người thanh toán hóa đơn
- `smoker`: có hút thuốc hay không
- `day`: ngày trong tuần
- `time`: thời gian dùng bữa(buổi sáng, buổi trưa, buổi tối)
- `size`: số lượng người tham gia

Bạn hãy thực hiện thống kê mô tả để tìm hiểu về hành vi tip của khách hàng tại nhà hàng

1. Đọc dữ liệu


In [61]:
# Your code here
import pandas as pd
import numpy as np
tip = pd.read_csv('data/tips.csv', header=0)
tip.head(10)

Unnamed: 0  total_bill   tip     sex smoker  day    time  size
0           0       16.99  1.01  Female     No  Sun  Dinner     2
1           1       10.34  1.66    Male     No  Sun  Dinner     3
2           2       21.01  3.50    Male     No  Sun  Dinner     3
3           3       23.68  3.31    Male     No  Sun  Dinner     2
4           4       24.59  3.61  Female     No  Sun  Dinner     4
5           5       25.29  4.71    Male     No  Sun  Dinner     4
6           6        8.77  2.00    Male     No  Sun  Dinner     2
7           7       26.88  3.12    Male     No  Sun  Dinner     4
8           8       15.04  1.96    Male     No  Sun  Dinner     2
9           9       14.78  3.23    Male     No  Sun  Dinner     2

2. Xoá cột `Unnamed: 0`


In [62]:
# Your code here
tip = tip.drop('Unnamed: 0', axis = 1)
tip.head(10)

total_bill   tip     sex smoker  day    time  size
0       16.99  1.01  Female     No  Sun  Dinner     2
1       10.34  1.66    Male     No  Sun  Dinner     3
2       21.01  3.50    Male     No  Sun  Dinner     3
3       23.68  3.31    Male     No  Sun  Dinner     2
4       24.59  3.61  Female     No  Sun  Dinner     4
5       25.29  4.71    Male     No  Sun  Dinner     4
6        8.77  2.00    Male     No  Sun  Dinner     2
7       26.88  3.12    Male     No  Sun  Dinner     4
8       15.04  1.96    Male     No  Sun  Dinner     2
9       14.78  3.23    Male     No  Sun  Dinner     2

3. Cho biết dữ liệu có bao nhiêu dòng, bao nhiêu thuộc tính.


In [10]:
# Your code here
tip.shape

(244, 7)

4. Trong các thuộc tính, thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng


In [64]:
# Your code here
tip.dtypes

total_bill    float64
tip           float64
sex            object
smoker         object
day            object
time           object
size            int64
dtype: object

5. Có bao nhiêu nam, bao nhiêu nữ. Tỷ lệ bao nhiêu?


In [85]:
female = tip['sex'].value_counts()['Female']
countt = tip['sex'].count()
print(f"Nữ: {female*100/countt} %")

Nữ: 35.65573770491803 %


In [84]:
# Your code here
female = tip['sex'].value_counts()['Female']
male = tip['sex'].value_counts()['Male']
countt = tip['sex'].count()
print("Số nam: ", male)
print("Tỉ lệ: ", round(male *100 / countt, 2), "%")
print("Số nữ: ", female)
print("Tỉ lệ: ", round(female *100 / countt, 2), "%")

Số nam:  157
Tỉ lệ:  64.34 %
Số nữ:  87
Tỉ lệ:  35.66 %


6. Có bao nhiêu người hút thuốc. Tỷ lệ bao nhiêu?


In [48]:
# Your code here
yes = tip['smoker'].value_counts()['Yes']
countt = tip['smoker'].count()
print("Số người hút thuốc: ", yes)
print("Tỉ lệ: ", round(yes*100 / summ, 2), "%")

Số người hút thuốc:  93
Tỉ lệ:  38.11 %


7. Ngày nào trong tuần khách hàng hay đến dùng bữa nhất. Tỷ lệ bao nhiêu?

In [91]:
# Đếm số lượng mỗi ngày
day_counts = tip['day'].value_counts()

# Tìm ngày có số lượng lớn nhất
most_frequent_day = day_counts.idxmax()

# Tính tỷ lệ
ratio = day_counts.max() *100 / day_counts.sum()

print(f"Ngày khách hàng hay đến dùng bữa nhất là: {most_frequent_day}, tỷ lệ: {round(ratio,2)}%")

Ngày khách hàng hay đến dùng bữa nhất là: Sat, tỷ lệ: 35.66%


In [70]:
# Your code here
tip_day = tip.groupby("day").size()
tip_day = tip_day.sort_values(ascending=False).head(1)
maxx = tip_day.max()
summ = tip['day'].count()
print("Ngày khách hàng hay đến dùng bữa nhất: ")
print(tip_day)
print("Tỉ lệ: ", round(maxx*100/summ), "%")

Ngày khách hàng hay đến dùng bữa nhất: 
day
Sat    87
dtype: int64
Tỉ lệ:  36 %


8. Nhà hàng phục vụ bữa ăn nào nhiều nhất (Breakfast, Lunch, Dinner)?

In [92]:
timee = tip['time'].value_counts()
time_max = timee.idxmax()
print(time_max)

Dinner


In [84]:
# Your code here
tip_time = tip.groupby("time").size()
tip_time = tip_time.sort_values(ascending=False).head(1)
print("Nhà hàng phục vụ bữa ăn nhiều nhất: ")
print(tip_time)

Nhà hàng phục vụ bữa ăn nhiều nhất: 
time
Dinner    176
dtype: int64


9. Nhóm khách hàng dùng bữa tại nhà hàng nhiều nhất là bao nhiêu người?


In [93]:
sizee = tip['size'].value_counts()
group = sizee.idxmax()
group

2

In [96]:
# Your code here
tip_size = tip.groupby("size").size()
tip_size= tip_size.sort_values(ascending=False).head(1)
print("Nhóm khách hàng dùng bữa tại nhà hàng nhiều nhất: ")
print(tip_size)

Nhóm khách hàng dùng bữa tại nhà hàng nhiều nhất: 
size
2    156
dtype: int64


10. Trung bình trong một bữa ăn nhà hàng phục vụ mấy khách?

In [96]:
# Tính số lượng khách trung bình trong một bữa ăn
average_guests_per_meal = tip['size'].mean()

print(f"Trung bình trong một bữa ăn, nhà hàng phục vụ {round(average_guests_per_meal,2)} khách.")

Trung bình trong một bữa ăn, nhà hàng phục vụ 2.57 khách.


In [110]:
# Your code here
sizee = tip['size'].sum()
sizeee = tip['time'].count()
print(sizee / sizeee)

2.569672131147541


---